tests/bases/test_ddp.py

```diff
@@ -116,3 +119,54 @@ def compute(self):
 def test_non_contiguous_tensors():
     """ Test that gather_all operation works for non contiguous tensors """
     torch.multiprocessing.spawn(_test_non_contiguous_tensors, args=(2, ), nprocs=2)
+
+
+def _test_state_dict_is_synced(rank, worldsize, tmpdir):
+    setup_ddp(rank, worldsize)
+
+    class DummyCatMetric(Metric):
+
+        def __init__(self):
+            super().__init__()
+            self.add_state("x", torch.tensor(0), dist_reduce_fx=torch.sum)
+            self.add_state("c", torch.tensor(0), dist_reduce_fx=torch.sum)
+
+        def update(self, x):
+            self.x += x
+            self.c += 1
+
+        def compute(self):
+            return self.x / self.c
+
+    metric = DummyCatMetric()
+    metric.persistent(True)
+
+    for i in range(5):
+        metric(i)
+        state_dict = metric.state_dict()
+
+        assert state_dict["x"] == sum(range(i + 1)) * 2
+        assert metric.x == sum(range(i + 1))
```

Comment:
sum(range(n + 1)) = n * (n + 1) / 2

```suggestion
        assert metric.x == steps * (steps + 1) / 2
        assert state_dict["x"] == metric.x * 2
```

Reply:
ahahah :) Done !